# 
# Hotplug support patch for mv643xx_eth driver used on pegasos machines.
# Author: Nicolas Det <nicolas.det@bplan-gmbh.de>
# Upstream status: In the process of being submitted, may need a bit of
# cleanup in order to not break embedded arches using this controller, but
# should not be a worry for debian.
#
--- drivers/net/mv643xx_eth.c.orig	2006-01-01 17:22:07.000000000 +0000
+++ drivers/net/mv643xx_eth.c	2006-01-01 17:23:56.000000000 +0000
@@ -41,6 +41,8 @@
 #include <linux/ethtool.h>
 #include <linux/platform_device.h>
 
+#include <linux/pci.h>
+
 #include <asm/io.h>
 #include <asm/types.h>
 #include <asm/pgtable.h>
@@ -1635,6 +1637,15 @@
 		" and Dale Farnsworth");
 MODULE_DESCRIPTION("Ethernet driver for Marvell MV643XX");
 
+static struct pci_device_id pci_marvell_mv64360[] = {
+	{ PCI_DEVICE(PCI_VENDOR_ID_MARVELL, PCI_DEVICE_ID_MARVELL_MV64360) },
+	{ PCI_DEVICE(PCI_VENDOR_ID_MARVELL, PCI_DEVICE_ID_MARVELL_MV64460) },
+	{ }
+};
+
+MODULE_DEVICE_TABLE(pci, pci_marvell_mv64360);
+
+
 /*
  * The second part is the low level driver of the gigE ethernet ports.
  */

